File: Data Structures & Algorithms/validate-parentheses/submission-2.py
class Solution:
    def isValid(self, s: str) -> bool:
        st = []
        for i in s:
            if i == '[' or i == '{' or i == '(':
                st.append(i)
            elif i == ']' and st and st[-1] == '[':
                st.pop()
            elif i == '}' and st and st[-1] == '{':
                st.pop()
            elif i == ')' and st and st[-1] == '(':
                st.pop()
            else:
                st.append(i)
        if len(st) == 0:
            return True
        return False

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer>st=new Stack<>();
        for(String i : tokens){
            int a;
            int b;
            int num;
            switch(i){
                case("+"):
                 a=st.pop();
                 b=st.pop();
                st.push(a+b);
                break;
                case("-"):
                 a = st.pop();
                 b = st.pop();
                 num = b-a;
                st.push(num);
                break;
                case("*"):
                 a = st.pop();
                 b = st.pop();
                 num = a*b;
                 st.push(num);
                 break;
                case("/"):
                 a = st.pop();
                 b = st.pop();
                 num = b/a;
                st.push(num);
                break;
                default:
                st.push(Integer.parseInt(i));
                break;
            }
        }
        return st.pop();
    }
}
